### PR TITLE
chore: set managed user locale via api v2

### LINKED ISSUE #15688: [CAL-4002] [ApiV2] enable setting locale field for managed user
Post and Patch managed user should enable updating the locale of a user, locale is set to "en" by default, class validator should make sure we are not passing invalid values

check: 
/apps/api/v2/src/modules/users/inputs/update-managed-user.input.ts
/apps/api/v2/src/modules/users/inputs/create-managed-user.input.ts
PATCH apps/api/v2/src/ee/me/me.controller.ts
POST /apps/api/v2/src/modules/oauth-clients/controllers/oauth-client-users/oauth-client-users.controller.ts

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-4002](https://linear.app/calcom/issue/CAL-4002/[apiv2]-enable-setting-locale-field-for-managed-user)</sub>

### PR DESCRIPTION
**Pull request by ThyMinimalDev**
## What does this PR do?

- Fixes #15688
- Fixes [CAL-4002](https://linear.app/calcom/issue/CAL-4002/[apiv2]-enable-setting-locale-field-for-managed-user)

Managed user locale can now be set via API

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

yarn test:e2e oauth-client-users.controller.e2e-spec.ts

### CODE REVIEW

File: apps/api/v2/src/modules/oauth-clients/controllers/oauth-client-users/outputs/managed-user.output.ts [hunk 1]
```diff
@@ -26,4 +28,9 @@ export class ManagedUserOutput {
 
   @ApiProperty({ example: null })
   defaultScheduleId!: number | null;
+
+  @IsEnum(Locales)
+  @IsOptional()
+  @ApiProperty({ example: Locales.EN, enum: Locales })
+  locale?: Locales;
```
Comment (supalarry) [inline apps/api/v2/src/modules/oauth-clients/controllers/oauth-client-users/outputs/managed-user.output.ts:35]: We can define locale type as:
```
locale!: keyof typeof Locales;
```

File: apps/api/v2/src/modules/users/inputs/create-managed-user.input.ts [hunk 2]
```diff
@@ -30,4 +31,9 @@ export class CreateManagedUserInput {
   @CapitalizeTimeZone()
   @ApiProperty({ example: "America/New_York" })
   timeZone?: string;
+
+  @IsEnum(Locales)
+  @IsOptional()
+  @ApiProperty({ example: Locales.EN, enum: Locales })
+  locale?: Locales;
```
Comment (supalarry) [inline apps/api/v2/src/modules/users/inputs/create-managed-user.input.ts:38]: We can define locale type as:

```
locale!: keyof typeof Locales;
```

File: apps/api/v2/src/modules/users/inputs/update-managed-user.input.ts [hunk 3]
```diff
@@ -34,4 +35,9 @@ export class UpdateManagedUserInput {
   @IsOptional()
   @CapitalizeTimeZone()
   timeZone?: string;
+
+  @IsEnum(Locales)
+  @IsOptional()
+  @ApiProperty({ example: Locales.EN, enum: Locales })
+  locale?: Locales;
```
Comment (supalarry) [inline apps/api/v2/src/modules/users/inputs/update-managed-user.input.ts:42]: We can define locale type as:

```
locale!: keyof typeof Locales;
```

**Review by supalarry — COMMENTED** [review]
Looks good, just small note below:

File: apps/api/v2/src/modules/users/inputs/create-managed-user.input.ts (hunk 2 above)
Comment (ThyMinimalDev) [reply apps/api/v2/src/modules/users/inputs/create-managed-user.input.ts:38]: not sure to understand why ?

Comment (supalarry) [reply apps/api/v2/src/modules/users/inputs/create-managed-user.input.ts:38]: An example would be when typing a test object in e2e

**Review by supalarry — APPROVED** [review]

**Pull request merged**